app-games/src/main/java/com/aptoide/android/aptoidegames/play_and_earn/presentation/permissions/OverlayPermissionActivity.kt

```diff
@@ -0,0 +1,77 @@
+package com.aptoide.android.aptoidegames.play_and_earn.presentation.permissions
+
+import android.app.AppOpsManager
+import android.content.Intent
+import android.os.Bundle
+import android.provider.Settings
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.net.toUri
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class OverlayPermissionActivity : AppCompatActivity() {
+
+  private lateinit var appOps: AppOpsManager
+
+  private var permissionCheckJob: Job? = null
+  private val checkInterval = 600L
+
+  val overlayPermissionLauncher =
+    registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+      bringAppToForeground()
+    }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    appOps = getSystemService(APP_OPS_SERVICE) as AppOpsManager
+
+    startPermissionCheck()
+
+    overlayPermissionLauncher.launch(
+      Intent(
+        Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+        ("package:$packageName").toUri()
+      )
+    )
+  }
+
+  fun launchPermission() {
```

Comment:
I would name this method differently since the permission it is actually launching is the usage stats and not the overlay one. Since it doesnt have the specific name and we are on the overlay class this could be missinterpreted